src/server/game/Spells/Spell.cpp

```diff
@@ -3145,6 +3145,11 @@ SpellCastResult Spell::prepare(SpellCastTargets const& targets, AuraEffect const
         if (!(_triggeredCastFlags & TRIGGERED_IGNORE_GCD))
             TriggerGlobalCooldown();
 
+        // Call CreatureAI hook OnSpellStart
+        if (Creature* caster = m_originalCaster->ToCreature())
```

Comment:
it might be useful to write a comment in the code about why m_originalCaster is used instead of m_caster

Reply:
@offl this was not solved. The PR should not have been merged yet.

Please change this to m_caster. m_originalCaster can be null, see https://github.com/offl/TrinityCore/blob/b3ecef24e37e3ed4f424acdb4b86de05e09d697a/src/server/game/Spells/Spell.cpp#L545 , and it can cause crashes.

See https://gist.github.com/Aokromes/c443ab3b8f49cec509a2938e98ceffbe

@Aokromes tagging you